src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+     * The 'with' array is used in conjunction with the 'translate' component.
+     *
+     * @return The with elements.
+     */
+    public List<JsonElement> getWith() {
+        return Collections.unmodifiableList(with);
+    }
+
+    /**
+     * Adds an element to the 'with' array.
+     *
+     * @param element The JSON element.
+     *
+     * @return This object.
+     */
+    public ChatComponent addWith(JsonElement element) {
+        $lock.writeLock().lock();
+        with.add(element);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Adds an string to the 'with' array.
+     *
+     * @param element The string.
+     *
+     * @return This object.
+     */
+    public ChatComponent addWith(String element) {
+        return addWith(new JsonPrimitive(element));
+    }
+
+    /**
+     * Gets all components attached to the 'extra' array.
+     *
+     * Components in this array are appended to this component.
+     *
+     * @return The extra components.
+     */
+    public List<ChatComponent> getExtra() {
+        return Collections.unmodifiableList(extra);
+    }
+
+    /**
+     * Adds a component to the 'extra' array.
+     *
+     * @param component The component.
+     *
+     * @return This component.
+     */
+    public ChatComponent addExtra(ChatComponent component) {
+        if (!hasExtra(component, true)) {
+            $lock.writeLock().lock();
+            this.extra.add(component);
+            $lock.writeLock().unlock();
+        }
+        return this;
+    }
+
+    /**
+     * Adds a string to the 'extra' array.
+     *
+     * @param string The string.
+     *
+     * @return This component.
+     */
+    public ChatComponent addExtra(String string) {
+        $lock.writeLock().lock();
+        this.extra.add(ChatComponent.create().setText(string));
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Checks if this component contains the given component, optionally recursively.
+     *
+     * @param component The component
+     * @param recursive Whether to check children as well.
+     *
+     * @return True iff the given component exists in this component's heirarchy.
+     */
+    public boolean hasExtra(ChatComponent component, boolean recursive) {
+        $lock.readLock().lock();
+        if (extra.contains(component)) {
+            $lock.readLock().unlock();
+            return true;
+        }
+        if (!recursive) {
+            $lock.readLock().unlock();
+            return false;
+        }
+        for (ChatComponent child : extra) {
+            if (child.hasExtra(component, true)) {
+                $lock.readLock().unlock();
+                return true;
+            }
+        }
+        $lock.readLock().unlock();
+        return false;
+    }
+
+    /**
+     * Gets whether or not this component is bold.
+     *
+     * @return True iff it is.
+     */
+    public boolean isBold() {
+        return bold != null && bold.get();
```

Comment:
THREAD-SAFETY
This is a poor construct to use for a thread-safe class. It doesn't save (that much) memory, and can be solved by using a volatile boolean set to false as a default value instead.